src/backend/impress/urls.py

```diff
@@ -1,22 +1,55 @@
 """URL configuration for the impress project"""
 
+import os
+
 from django.conf import settings
 from django.conf.urls.static import static
 from django.contrib import admin
 from django.contrib.staticfiles.urls import staticfiles_urlpatterns
 from django.urls import include, path, re_path
 
+from django_prometheus import exports
 from drf_spectacular.views import (
     SpectacularJSONAPIView,
     SpectacularRedocView,
     SpectacularSwaggerView,
 )
 
+from core.api.custom_probe_views import liveness_check, readiness_check
+from core.api.decorators import monitoring_cidr_protected_view
+
 urlpatterns = [
     path("admin/", admin.site.urls),
     path("", include("core.urls")),
 ]
 
+# Conditionally add Prometheus Exporter endpoint
+if os.environ.get("MONITORING_PROMETHEUS_EXPORTER", "False").lower() == "true":
+    urlpatterns.append(
+        path(
+            "prometheus/",
+            monitoring_cidr_protected_view(exports.ExportToDjangoView),
+            name="prometheus-django-metrics",
+        ),
+    )
+
+# Conditionally add liveness and readiness probe endpoints
+if os.environ.get("MONITORING_PROBING", "False").lower() == "true":
+    urlpatterns.append(
+        path(
+            "probes/liveness/",
+            monitoring_cidr_protected_view(liveness_check),
+            name="liveness-probe",
+        ),
+    )
+    urlpatterns.append(
+        path(
+            "probes/readiness/",
+            monitoring_cidr_protected_view(readiness_check),
+            name="readiness-probe",
+        ),
+    )
```

Comment:
My understanding is that this would be redundant with what we already have on /__lbheartbeat__ and /__heartbeat__ with [Mozilla's dockerflow](https://github.com/mozilla-services/Dockerflow). 
Can you check before I review the rest?

Reply:
One yes, the other not so much.

The liveness probe `liveness_check` is essentially the same as the dockerflow heartbeat.
But the readiness probe `readiness_check` reaches more deeply into the application, checking DB, cache and media backend before returning 200 so it catches failures Dockerflow would miss.

What to do?

Reply:
@lunika can you help me with the decision?